### PR TITLE
style: optimize button code

### PR DESCRIPTION
**Pull request by ayangweb**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [x] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     optimize button code      |
| 🇨🇳 Chinese |      优化按钮组件代码     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 002ba37</samp>

Enhanced the `Button` component's TypeScript support and code quality by fixing type issues and removing redundant code in `components/button/button.tsx`.

### <samp>🤖 Generated by Copilot at 002ba37</samp>

*  Type `loadingOrDelay` as `LoadingConfigType` to avoid implicit any errors and improve type safety ([link](https://github.com/ant-design/ant-design/pull/42005/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L123-R123)).

### CODE REVIEW

**Review by li-jia-nan — APPROVED** [review]

File: components/button/button.tsx [hunk 1]
```diff
@@ -120,13 +120,16 @@ const InternalButton: React.ForwardRefRenderFunction<
...
-  const loadingOrDelay: LoadingConfigType = React.useMemo(
+  const loadingOrDelay = React.useMemo<LoadingConfigType>(
     () => getLoadingConfig(loading),
     [loading],
   );
   const [innerLoading, setLoading] = React.useState<Loading>(loadingOrDelay.loading);
   const [hasTwoCNChar, setHasTwoCNChar] = React.useState(false);
-  const buttonRef = (ref as any) || React.createRef<HTMLAnchorElement | HTMLButtonElement>();
+
+  const internalRef = React.createRef<HTMLAnchorElement | HTMLButtonElement>();
```
Comment (li-jia-nan) [inline components/button/button.tsx:130]: 这个 `createRef` 是不是可以用 `useRef` 代替呢？

Comment (ayangweb) [reply components/button/button.tsx:130]: #42008

File: components/button/button.tsx [hunk 2]
```diff
@@ -147,10 +150,10 @@ const InternalButton: React.ForwardRefRenderFunction<
   };
 
   React.useEffect(() => {
-    let delayTimer: number | null = null;
+    let delayTimer: NodeJS.Timer | null = null;
```
Comment (crazyair) [inline components/button/button.tsx:153]: 类型不应该用`ReturnType<typeof setTimeout>` 吗？cc @zombieJ

Comment (MadCcc) [reply components/button/button.tsx:153]: 这是 4 月的 PR 了，现在应该都改了

**Pull request merged**